homeassistant/components/fjaraskupan/__init__.py

```diff
@@ -66,15 +72,25 @@ async def _async_refresh(
     async def _async_update_data(self) -> State:
         """Handle an explicit update request."""
         if self._refresh_was_scheduled:
-            raise UpdateFailed("No data received within schedule.")
+            if async_address_present(self.hass, self.device.address):
+                return self.device.state
+            raise UpdateFailed(
+                "No data received within schedule, and device is no longer present"
+            )
 
         await self.device.update()
         return self.device.state
 
-    def detection_callback(
-        self, ble_device: BLEDevice, advertisement_data: AdvertisementData
-    ) -> None:
+    def detection_callback(self, service_info: BluetoothServiceInfo) -> None:
         """Handle a new announcement of data."""
+        ble_device = BLEDevice(
+            service_info.address, service_info.name, service_info.rssi
+        )
+        advertisement_data = AdvertisementData(
+            local_name=service_info.name,
+            manufacturer_data=service_info.manufacturer_data,
+        )
+
```

Comment:
This is not really needed, just that the current lib api require the objects right now.